packages/x-date-pickers/src/internals/hooks/useMobilePicker/useMobilePicker.types.ts

```diff
@@ -51,7 +53,7 @@ export interface ExportedUseMobilePickerSlotProps<
     {},
     PickerOwnerState
   >;
-  textField?: SlotComponentProps<typeof TextField, {}, Record<string, any>>;
+  textField?: SlotComponentPropsFromProps<PickersTextFieldProps, {}, FieldOwnerState>;
```

Comment:
Always great to have this kind of cleaning :ok_hand: 
Now the `slots` exposed at the picker level will always have the right `ownerState` as well